presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java

```diff
@@ -195,6 +195,14 @@ public void tearDown(Method method)
         log.info("FINISHED %s in %s verified %s expressions", method.getName(), Duration.nanosSince(start), futures.size());
     }
 
+    // This is a setter and not a test method.
+    // TestNG considers this as a test method since we have annotated the class with @Test.
+    @Test(enabled = false)
```

Comment:
The field `functionAssertions` is private, hence a no argument `setFunctionAssertions()` wont' be able to set `this.functionAssertions`. Instead of changing the access to public, instead I created the function `setFunctionAssertions(FunctionAssertions functionAssertions)` and calling it from base and native class.

Reply:
I think I meant make this a method that returns the desired functionAssertion that gets overridden in the native implementation.  e.g. `public FunctionAssertions createFunctionAssertions()`, and then in the beforeClass method do`functionAssertions = createFunctionAssertions()`.  Seems better than making a public setter method.  

Either way, shouldn't need the `@Test` annotation

Reply:
Sounds good.
`public FunctionAssertions createFunctionAssertions` does not require the @Test annotation, I think a public method with void return type was automatically being considered by the framework due to class level annotation @Test.